mm2src/mm2_main/src/lp_swap/swap_v2_common.rs

```diff
@@ -371,3 +373,99 @@ pub(super) async fn swap_kickstart_handler<
         error!("Error {} on trying to run the swap {}", e, uuid);
     }
 }
+
+pub(super) async fn swap_kickstart_handler_for_maker(
+    ctx: MmArc,
+    swap_repr: MakerSwapDbRepr,
+    storage: MakerSwapStorage,
+    uuid: Uuid,
+) {
+    let (maker_coin, taker_coin) = match swap_kickstart_coins(&ctx, &swap_repr, &uuid).await {
+        Some(coins) => coins,
+        None => return,
+    };
```

Comment:
Both `swap_kickstart_handler_for_taker` and `swap_kickstart_handler_for_maker` depending on this initial logic, what about not spawning these functions at all if this is `None` from the parent function? This way we could fetch this data only once (instead of doing it twice) and avoid spawning futures for no reason.

Reply:
> This way we could fetch this data only once (instead of doing it twice)

what do you mean? TPU is not symmetric like legacy protocol. 
Taker state machine has its own taker-maker coins, Maker state machine has its own as well.

Reply:
Im not sure its a good idea to move `lp_coinfind` out of the future spawn.  
I think in the original code, coins find logic is intentionally part of the spawned future (in both legacy and TPU) to find coins tickers that must be enabled for swaps, spawn tasks, return coins tickers, and continue the app init process. 
If `lp_coinfind` returns `Ok(None)`, then `swap_kick_starts` function will get stuck at this point until the coin is activated (we have a `loop` cycle).  

Also, according to this comment, I assume the enable RPC even won't be available:
https://github.com/KomodoPlatform/komodo-defi-framework/blob/676d09c4b574f9f20d7e7f16e1097c7d7427d5cd/mm2src/mm2_main/src/lp_native_dex.rs#L483-L485

Reply:
Oops, I thought they were sharing the exact same query.